From: NeilBrown <neilb@suse.de>
Date: Fri, 7 Feb 2020 11:05:03 -0500
Subject: Allow compilation to succeed with -fno-common
Origin: https://git.linux-nfs.org/?p=steved/nfs-utils.git;a=commit;h=1eafe05616b4660fb15d106e06f0b3e18983708c
Bug-Debian: https://bugs.debian.org/957608

When compiled with -fno-common, global variables that are declared
multple times cause an error.  With -fcommon (the default), they are
merged.

Declaring such variable multiple times is probably not a good idea, and
is definitely not necessary.

This patch changes all the global variables defined in include files to
be explicitly "extern", and where necessary, adds the variable
declaration to a suitable .c file.

To test, run
  CFLAGS=-fno-common ./configure
  make

Signed-off-by: NeilBrown <neilb@suse.de>
Signed-off-by: Steve Dickson <steved@redhat.com>
[Salvatore Bonaccorso: Backport to 1.3.4:
 - Drop changes in utils/nfsdcld/
 - support/nfs/conffile.c: Drop definition of conf_path
]
---
 utils/mountd/v4root.c        |  2 --
 utils/nfsdcld/cld-internal.h | 10 +++++-----
 utils/nfsdcld/nfsdcld.c      |  6 ++++++
 utils/statd/statd.c          |  1 +
 utils/statd/statd.h          |  2 +-
 5 files changed, 13 insertions(+), 8 deletions(-)

--- a/utils/mountd/v4root.c
+++ b/utils/mountd/v4root.c
@@ -28,8 +28,6 @@
 #include "v4root.h"
 #include "pseudoflavors.h"
 
-int v4root_needed;
-
 static nfs_export pseudo_root = {
 	.m_next = NULL,
 	.m_client = NULL,
--- a/utils/statd/statd.c
+++ b/utils/statd/statd.c
@@ -66,6 +66,7 @@ static struct option longopts[] =
 };
 
 extern void sm_prog_1 (struct svc_req *, register SVCXPRT *);
+stat_chge	SM_stat_chge;
 
 #ifdef SIMULATIONS
 extern void simulator (int, char **);
--- a/utils/statd/statd.h
+++ b/utils/statd/statd.h
@@ -41,7 +41,7 @@ extern void	load_state(void);
 /*
  * Host status structure and macros.
  */
-stat_chge		SM_stat_chge;
+extern stat_chge	SM_stat_chge;
 #define MY_NAME		SM_stat_chge.mon_name
 #define MY_STATE	SM_stat_chge.state
 
--- a/support/nfs/conffile.c
+++ b/support/nfs/conffile.c
@@ -104,7 +104,6 @@ struct conf_binding {
   int is_default;
 };
 
-char *conf_path;
 LIST_HEAD (conf_bindings, conf_binding) conf_bindings[256];
 
 static char *conf_addr;
